indico/web/client/js/react/components/PersonLinkField.jsx

```diff
@@ -0,0 +1,400 @@
+    </div>
+    <div styleName="actions">
+      <Icon
+        styleName="button edit"
+        name="pencil alternate"
+        size="large"
+        onClick={onEdit}
+        disabled={disabled}
+      />
+      {canDelete && (
+        <Icon
+          styleName="button delete"
+          name="remove"
+          size="large"
+          onClick={onDelete}
+          disabled={disabled}
+        />
+      )}
+    </div>
+  </PrincipalItem>
+);
+
+PersonListItem.propTypes = {
+  person: PropTypes.object.isRequired,
+  roles: PropTypes.array.isRequired,
+  onDelete: PropTypes.func,
+  canDelete: PropTypes.bool,
+  onEdit: PropTypes.func.isRequired,
+  disabled: PropTypes.bool,
+  avatarURL: PropTypes.string,
+  onClickRole: PropTypes.func,
+};
+
+PersonListItem.defaultProps = {
+  canDelete: true,
+  disabled: false,
+  avatarURL: null,
+  onDelete: null,
+  onClickRole: null,
+};
+
+const PersonLinkSection = ({
+  label: sectionLabel,
+  persons,
+  defaultRoles,
+  onChange,
+  onEdit,
+  canDelete,
+}) => {
+  const onClickRole = (personIndex, roleIndex, value) => {
+    const role = defaultRoles[roleIndex];
+    const roles = value
+      .filter((r, i) =>
+        // ensure only a single section is selected at a time
+        roleIndex === i ? !r.active : r.active && (!role.section || !defaultRoles[i].section)
+      )
+      .map(r => r.name);
+    onChange(persons.map((v, i) => (i === personIndex ? {...v, roles} : v)));
+  };
+
+  return (
+    <>
+      {sectionLabel && <div styleName="titled-rule">{sectionLabel}</div>}
+      <List divided relaxed>
+        {persons.length > 0 ? (
+          persons.map((p, idx) => (
+            <PersonListItem
+              key={p.userId || p.email}
+              person={p}
+              onDelete={() => onChange(persons.filter((_, i) => i !== idx))}
+              onEdit={() => onEdit(idx)}
+              onClickRole={(roleIdx, value) => onClickRole(idx, roleIdx, value)}
+              canDelete={canDelete}
+              roles={defaultRoles.map(({name, ...rest}) => ({
+                ...rest,
+                name,
+                active: p.roles && p.roles.includes(name),
+              }))}
+            />
+          ))
+        ) : (
+          <Translate>There are no persons</Translate>
+        )}
+      </List>
+    </>
+  );
+};
+
+PersonLinkSection.propTypes = {
+  label: PropTypes.string,
+  persons: PropTypes.array,
+  defaultRoles: PropTypes.array,
+  onChange: PropTypes.func.isRequired,
+  onEdit: PropTypes.func.isRequired,
+  canDelete: PropTypes.bool,
+};
+
+PersonLinkSection.defaultProps = {
+  label: undefined,
+  persons: [],
+  defaultRoles: [],
+  canDelete: true,
+};
+
+// TODO: each person includes way more data than needed in the form data
+export default function PersonLinkField({
+  value: persons,
+  onChange,
+  eventId,
+  sessionUser,
+  roles,
+  emptyMessage,
+}) {
+  const [favoriteUsers] = useFavoriteUsers();
+  const [modalOpen, setModalOpen] = useState(false);
```

Comment:
Not needed; `selected !== null` covers it already.

Reply:
But we might not have it while manually inserting a person.

```
<Translate as={Button} type="button" onClick={() => setModalOpen(true)}>
    Enter manually
</Translate>
```

Reply:
ah, missed that one...